### PR TITLE
Added an option to include some visual feedback for scanned qr codes.

### PR DESCRIPTION
**Pull request by howech**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -30,23 +30,66 @@ module.exports = class Reader extends Component {
...
+    constraints: null,
+    showFeedback: false,
   };
 
   els = {};
 
+  // This data gets dynamically updated - its used to transform
+  // coordinates in the video image into coordinates in css %-space
+  transformData = {
+    dx: 0,
```
Comment (dhruvbansal) [inline src/index.js]: I think `x_offset` and `y_offset` are better names for these variables.

File: src/index.js [hunk 2]
```diff
@@ -30,23 +30,66 @@ module.exports = class Reader extends Component {
...
+      width: new_width + "%",
+    }
+  }
+
   constructor(props) {
     super(props)
 
     this.state = {
       mirrorVideo: false,
+      qrFound: false,
```
Comment (dhruvbansal) [inline src/index.js:92]: `handleWorkerMessage` below will add `top`, `left`, &c. to `this.state` all with value `0`.  I think you should initialize `this.state` the same way, if only as a form of documentation.

File: src/index.js [hunk 3]
```diff
@@ -353,6 +448,16 @@ module.exports = class Reader extends Component {
...
 
+    const qrLocatorStyle = {
+      top: top,
+      left: left,
+      outline: '10px solid rgba(0, 255, 0, 0.5)',
+      position: 'absolute',
+      width: width,
+      height: height,
+      zIndex: 2,
+      visibility: qrFound ? 'visible' : 'hidden'
```
Comment (dhruvbansal) [inline src/index.js]: Instead of checking  just `qrFound` here you should check `qrFound && showFeedback`.

File: src/index.js [hunk 4]
```diff
@@ -268,9 +324,39 @@ module.exports = class Reader extends Component {
...
+      const org_right = Math.max(decoded.location.topRightCorner.x, decoded.location.bottomRightCorner.x)
+      const org_top = Math.min(decoded.location.topLeftCorner.y, decoded.location.topRightCorner.y)
+      const org_bottom = Math.max(decoded.location.bottomLeftCorner.y, decoded.location.bottomRightCorner.y)
+
+      // Transform the video-based coordinates into %-based units suitable
+      // for css on a div
+      const {top, left, width, height} = this.transformRect(org_top, org_left, org_bottom, org_right)
+
+      this.setState({
+        qrFound: showFeedback,
```
Comment (dhruvbansal) [inline src/index.js]: You should set `qrFound` to `true` here and let the display code below decide what to do with that fact.

File: src/index.js [hunk 5]
```diff
@@ -377,6 +482,11 @@ module.exports = class Reader extends Component {
               ? <img style={imgPreviewStyle} ref={this.setRefFactory('img')} onLoad={onImageLoad} />
               : <video style={videoPreviewStyle} ref={this.setRefFactory('preview')} />
           }
+          {
+            legacyMode
```
Comment (dhruvbansal) [inline src/index.js:491]: I can't decide if it's better to include this div and make it invisible (as you have done here) vs not include it at all unless it's necessary, something like:

`{legacyMode && qrFound && showFeedback && <div style={qrLocatorStyle} />}`

Even if you stick with the current approach, I think it's more readable to say 

`{legacyMode && <div style={qrLocatorStyle} />}`

File: src/index.js [hunk 6]
```diff
@@ -268,9 +324,39 @@ module.exports = class Reader extends Component {
...
+    } else {
+      this.setState({
+        qrFound: false,
+        top: 0,
+        left: 0,
+        width: 0,
+        height: 0,
+      })
+    }
+    onScan(decoded && decoded.data || null)
```
Comment (dhruvbansal) [inline src/index.js]: I think explicit parens here would be nice:

`onScan(decoded && (decoded.data || null))`

**Review by dhruvbansal — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 5 above)
Comment (howech) [reply src/index.js:491]: I like having it hidden, it made it a lot easier to debug some things.  While I dont much care for the legacyMode ? null : <div /> thing, it does match the style of the code above it, so I'm gonna leave this one as is.

**Review by dhruvbansal — APPROVED** [review]

**Comment by howech** [comment]
Rebased and squashed. See #2

**Pull request closed without merging**